src/main/java/games/strategy/engine/framework/startup/mc/ServerModel.java

```diff
@@ -213,24 +226,28 @@ private ServerProps getServerProps(final Component ui) {
       }
       return null;
     }
-    final ServerProps props = new ServerProps();
+    final ServerConnectionProps props = new ServerConnectionProps();
     props.setName(options.getName());
     props.setPort(options.getPort());
     props.setPassword(options.getPassword());
     return props;
   }
 
+
+  public boolean createServerMessenger(final Component ui) {
+    final ServerConnectionProps props = getServerProps(ui);
+    return props != null && createServerMessenger(ui, props);
+  }
+
   /**
    * UI can be null. We use it as the parent for message dialogs we show.
    * If you have a component displayed, use it.
    */
-  public boolean createServerMessenger(Component ui) {
-    ui = ui == null ? null : JOptionPane.getFrameForComponent(ui);
-    this.ui = ui;
-    final ServerProps props = getServerProps(ui);
-    if (props == null) {
-      return false;
-    }
+  boolean createServerMessenger(
+      @Nullable final Component ui,
+      @Nonnull final ServerConnectionProps props) {
```

Comment:
I'm not against using `@Nonnull`, in general.  However, in many projects I've worked on in the past, I've found that the absence of any nullity annotation implying "non-nullness" makes the code more readable.  A good goal is to avoid using nullable types anyway, so only using `@Nullable` makes it that much more obvious when a nullable type is present.

The Eclipse null analysis engine even uses a third annotation called `@NonNullByDefault`, which can be applied to indicate, for example, all types used in a package should be assumed to be non-nullable unless otherwise specified with a `@Nullable` attribute.  I believe they added it for the reason I mentioned above: simply to avoid all the noise of adding `@Nonnull` everywhere.  Does IntelliJ support a similar concept?

Granted, if we're really interested in enabling null analysis, applying `@Nonnull` in a local context like this is probably the right thing to do.  That way you can start null analysis in a very focused context and gradually expand it from methods to an entire class and then to an entire package.  So maybe just ignore this comment. :smile:

Reply:
Interesting to know about @NonNullByDefault. Some additional research there would be interesting, but at the same, given this code base we probably could not apply that as many values are nullable. Would be interesting to see how bad it is, but having something complaining of 100s of violations I don't think would be useful.

More importantly, the null params here is pretty ugly. This is mostly existing code and me hacking the code to make it a bit more friendly. Again, this PR stops at the point where it makes sense to throw away the old UI code so we can refactor the models to be sane. Ultimately we should not need such a constructor and should use the primary one, and not have any nullables fields. That all represents more/future work to be done IMO.

Reply:
Just for reference...  It turns out that the JSR 305 annotation library we use contains the `ParametersAreNonnullByDefault` and `ParametersAreNullableByDefault` annotations, which are similar but not exactly equivalent to Eclipse's `NonNullByDefault`.  The can be applied at different scopes (package, class, method), but they only apply to method parameters, whereas the Eclipse `NonNullByDefault` annotation also applies to fields and local variables.

But it's good to know they are there if we need them. :smile: